salesforcecdpconnector/metadata_processor.py

```diff
@@ -0,0 +1,151 @@
+                        genie_table_relationship.from_entity_attribute = relationship[
+                            RELATIONSHIP_FROM_ENTITY_ATTRIBUTE]
+                    if RELATIONSHIP_TO_ENTITY_ATTRIBUTE in relationship.keys():
+                        genie_table_relationship.to_entity_attribute = relationship[RELATIONSHIP_TO_ENTITY_ATTRIBUTE]
+                    if RELATIONSHIP_CARDINALITY in relationship.keys():
+                        genie_table_relationship.cardinality = relationship[RELATIONSHIP_CARDINALITY]
+
+                    genie_table_relationships.append(genie_table_relationship)
+                genie_table.relationships = genie_table_relationships
+
+            if GENIE_TABLE_INDEXES in table_metadata.keys():
+                genie_table.indexes = table_metadata[GENIE_TABLE_INDEXES]
+
+            fields = MetadataProcessor.__get_fields_of_genie_table(table_metadata)
+            genie_table.fields = fields
+            genie_table_list.append(genie_table)
+
+        return genie_table_list
+
+    @staticmethod
+    def __get_fields_of_genie_table(table_metadata):
+        genie_table_fields = []
+        if GENIE_TABLE_FIELDS in table_metadata.keys():
+            for field in table_metadata[GENIE_TABLE_FIELDS]:
+                genie_table_field = Field(field[FIELDS_NAME], field[FIELDS_DISPLAY_NAME], field[FIELDS_TYPE])
+                genie_table_fields.append(genie_table_field)
+
+        elif MetadataProcessor.__is_cio_genie_table(table_metadata[GENIE_TABLE_NAME]):
+            if GENIE_TABLE_DIMENSIONS in table_metadata.keys():
+                for dimension in table_metadata[GENIE_TABLE_DIMENSIONS]:
+                    genie_table_field_from_dimension = Field(dimension[FIELDS_NAME],
+                                                             dimension[FIELDS_DISPLAY_NAME],
+                                                             dimension[FIELDS_TYPE],
+                                                             is_measure=False, is_dimension=True)
+                    genie_table_fields.append(genie_table_field_from_dimension)
+
+            if GENIE_TABLE_MEASURES in table_metadata.keys():
+                for measure in table_metadata[GENIE_TABLE_MEASURES]:
+                    genie_table_field_from_measure = Field(measure[FIELDS_NAME],
+                                                           measure[FIELDS_DISPLAY_NAME],
+                                                           measure[FIELDS_TYPE],
+                                                           is_measure=True, is_dimension=False)
+                    genie_table_fields.append(genie_table_field_from_measure)
+
+        return genie_table_fields
+
+    @staticmethod
+    def __is_cio_genie_table(table_name):
+        if "cio" in table_name:
+            return True
+        return False
+
+    @staticmethod
+    @lru_cache()
+    def __describe_table_result(connection, request_params={}):
```

Comment:
request_params can be removed

Reply:
Request params are added back. Reason is that the caching is removed, and hence instead of filtering on client side, this need to be done in server side. To do that the request param is used